### ISSUE TITLE
AddNorm advanced features

### ISSUE THREAD

**Issue by soran-ghaderi** [tensorops/TransformerX#75]
- normalization: batch normalization, instance normalization, and layer normalization.
- specify an epsilon value for numerical stability in the normalization layer.
- residual connection with dropout.
- different activation functions.
- regularization of the kernel and bias weights.
- normalization techniques that can help to stabilize the training process and improve the performance of deep neural networks.
